From e407387926de4c75abd17bd1396caa95d35a4bea Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@xilinx.com>
Date: Mon, 17 Apr 2017 13:04:11 -0700
Subject: [PATCH] tools: Use POSIX poll.h instead of sys/poll.h

The POSIX spec specifies to use:
    #include <poll.h>
instead of:
    #include <sys/poll.h>
as seen here:
    http://pubs.opengroup.org/onlinepubs/009695399/functions/poll.html

This removes the warning:
    #warning redirecting incorrect #include <sys/poll.h> to <poll.h>
when building with the musl C-library.

Signed-off-by: Alistair Francis <alistair.francis@xilinx.com>
Acked-by: Wei Liu <wei.liu2@citrix.com>
Acked-by: Razvan Cojocaru <rcojocaru@bitdefender.com>
Acked-by: Ian Jackson <ian.jackson@eu.citrix.com>
Release-acked-by: Julien Grall <julien.grall@arm.com>
---
 tools/libxl/libxl_internal.h               | 2 +-
 tools/tests/xen-access/xen-access.c        | 2 +-
 tools/xenstat/libxenstat/src/xenstat_qmp.c | 2 +-
 tools/xentrace/xentrace.c                  | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tools/libxl/libxl_internal.h b/tools/libxl/libxl_internal.h
index be24b76dfa..5d082c5704 100644
--- a/tools/libxl/libxl_internal.h
+++ b/tools/libxl/libxl_internal.h
@@ -38,7 +38,7 @@
 #include <ctype.h>
 
 #include <sys/mman.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/select.h>
 #include <sys/stat.h>
 #include <sys/time.h>
diff --git a/tools/tests/xen-access/xen-access.c b/tools/tests/xen-access/xen-access.c
index ff4d289b45..238011e010 100644
--- a/tools/tests/xen-access/xen-access.c
+++ b/tools/tests/xen-access/xen-access.c
@@ -36,7 +36,7 @@
 #include <signal.h>
 #include <unistd.h>
 #include <sys/mman.h>
-#include <sys/poll.h>
+#include <poll.h>
 
 #include <xenctrl.h>
 #include <xenevtchn.h>
diff --git a/tools/xenstat/libxenstat/src/xenstat_qmp.c b/tools/xenstat/libxenstat/src/xenstat_qmp.c
index a87c9373c2..3fda487d49 100644
--- a/tools/xenstat/libxenstat/src/xenstat_qmp.c
+++ b/tools/xenstat/libxenstat/src/xenstat_qmp.c
@@ -14,7 +14,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/un.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/tools/xentrace/xentrace.c b/tools/xentrace/xentrace.c
index f09fe6cf19..364a6fdad5 100644
--- a/tools/xentrace/xentrace.c
+++ b/tools/xentrace/xentrace.c
@@ -24,7 +24,7 @@
 #include <getopt.h>
 #include <assert.h>
 #include <ctype.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/statvfs.h>
 
 #include <xen/xen.h>
-- 
2.11.0

